packages/sdk/src/mint/mint.ts

```diff
@@ -36,75 +35,91 @@ export const mint = (
     throw new Error(ERROR_MESSAGES.NO_MEDIA);
   }
 
-  if (splits) {
+  let adjustedRoyalties: Splits = {};
+  let royaltyTotal: number;
+
+  if (royalties) {
     // FIXME: suggest we use % (ints) vs float here
     // 0.5 -> 5000
-    adjustSplitsForContract(splits);
+    royaltyTotal = getRoyaltyTotal(royalties);
```

Comment:
Why iterate over royalties twice? Esp since we are calculating totals in both cases

Reply:
Ah, it's necessary for the adjustment